rpc_util.go

```diff
@@ -914,3 +923,91 @@ const (
 )
 
 const grpcUA = "grpc-go/" + Version
+
+const (
+	level0PoolMaxSize = 16
+	level1PoolMaxSize = level0PoolMaxSize * 16
+	level2PoolMaxSize = level1PoolMaxSize * 16 //   4 KB
+	level3PoolMaxSize = level2PoolMaxSize * 16 //  64 KB
+	level4PoolMaxSize = level3PoolMaxSize * 16 //   1 MB
+)
+
+type bytesPools struct {
```

Comment:
My thought process is as follows:
- We should have pools of different sizes, each one containing buffers of the same size.
- So, if we have three pools of size (10B, 100B, 1KB), all buffers within each of these pools should be of the same size.

The above approach may result in a bit more memory usage (because all requests for buffers of size 11-100 would result in a buffer of size 100 being returned), but would result in lesser number of allocations in general.

What do you think about this approach?

Reply:
As it's difficult to predict the ideal pool for every type of workload, I've decided to let users handle the implementation of the shared recv buffer pool themselves. This will allow each user to create a pool that suits their particular workload requirements.(For instance, I might opt for a pool that utilizes large, homogeneous memory.)

Reply:
As mentioned in my other comment, I do like your approach of defining an interface for this. But we should continue to provide a default implementation as well.